session/src/main/java/org/apache/iotdb/session/template/Template.java

```diff
@@ -0,0 +1,241 @@
+
+      cur = stack.pop();
+      count--;
+
+      if ((count < 0) && (childSize.size() > 0)) {
+        isAligned = alignStack.pop();
+        count = childSize.pop();
+        count--;
+        nowPaths.remove(nowPaths.size() - 1);
+      }
+
+      if (cur != moutedNode) nowPaths.add(cur.getName());
+
+      if (cur.getChildren() != null) {
+        childSize.push(count);
+        alignStack.push(isAligned);
+
+        for (Node child : cur.getChildren().values()) stack.push(child);
+        count = cur.getChildren().size();
+        isAligned = cur.isShareTime();
+      }
+
+      if (cur.isMeasurement()) {
+        mNode = (MeasurementNode) cur;
+        result.add(convertList2String(nowPaths));
+        measurementsPath.add(convertList2String(nowPaths));
+        alignedList.add(isAligned);
+        dataTypes.add(mNode.getDataType());
+        encodings.add(mNode.getEncoding());
+        compressors.add(mNode.getCompressionType());
+        nowPaths.remove(nowPaths.size() - 1);
+      }
+    }
+    measurementsPath = result;
+  }
+
+  public byte[] serialize() throws IOException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    Deque<Node> stack = new ArrayDeque<>();
+    stack.push(moutedNode);
+
+    while (stack.size() != 0) {
+      Node cur = stack.pop();
+      cur.serialize(baos);
+      if (!cur.isMeasurement()) {
+        for (Node child : cur.getChildren().values()) stack.push(child);
+      }
+    }
+    return baos.toByteArray();
+  }
+
+  // endregion
+
+  // region Inner Utility
+  public static void deserialize(ByteBuffer buffer) {
+    while (buffer.position() != buffer.limit()) {
+      String nodeName = readStringFromBuffer(buffer);
+      boolean isInternal = readBoolFromBuffer(buffer);
+
+      if (isInternal) {
+        boolean isAligned = readBoolFromBuffer(buffer);
+        int childSize = readIntFromBuffer(buffer);
+
+        System.out.println(
+            String.format("%s, %s, %s, %d", nodeName, isInternal, isAligned, childSize));
+      } else {
+        TSDataType dataType = TSDataType.values()[readByteFromBuffer(buffer)];
+        TSEncoding encoding = TSEncoding.values()[readByteFromBuffer(buffer)];
+        CompressionType compressor = CompressionType.values()[readByteFromBuffer(buffer)];
+
+        System.out.println(
+            String.format(
+                "%s, %s, %s, %s, %s", nodeName, isInternal, dataType, encoding, compressor));
+      }
+    }
+  }
+
+  private static int readIntFromBuffer(ByteBuffer buffer) {
+    return buffer.getInt();
+  }
+
+  private static String readStringFromBuffer(ByteBuffer buffer) {
+    int size = buffer.getInt();
+    byte[] bytes = new byte[size];
+    buffer.get(bytes, 0, size);
+    return new String(bytes, 0, size);
+  }
+
+  private static Byte readByteFromBuffer(ByteBuffer buffer) {
+    return buffer.get();
+  }
+
+  private static boolean readBoolFromBuffer(ByteBuffer buffer) {
+    return (buffer.get() == 1);
+  }
```

Comment:
these methods are already exist in ReadWriteIOUtils

Reply:
fixed with import ReadWriteIOUtils